src/Mod/Path/PathScripts/PathDressupLeadInOut.py

```diff
@@ -593,32 +568,62 @@ def generateLeadInOutCurve(self, obj):
         return Path.Path(commands)
 
 
-class TaskPanel:
+PROP_TYPE_QTYES = ["App::PropertyDistance", "App::PropertyAngle"]
+PROP_TYPE_NUMERIC = PROP_TYPE_QTYES + ["App::PropertyPercent", "App:PropertyFloat"]
+
+
+class SimpleEditPanel:
+    _fc = {}
+    obj = None
+    form = None
+
+    def getFields(self):
+        for prop_name, (get_field, set_field) in self._fc.items():
+            setattr(self.obj, prop_name, get_field())
+
+    def setFields(self):
+        for prop_name, (get_field, set_field) in self._fc.items():
+            set_field(getattr(self.obj, prop_name))
+
+    def connectWidget(self, prop_name, widget, custom_lbls={}):
+        prop_type = self.obj.getTypeIdOfProperty(prop_name)
+        widget_type = type(widget).__name__
+        if prop_type == "App::PropertyEnumeration" and widget_type == "QComboBox":
+            enum = self.obj.getEnumerationsOfProperty(prop_name)
+            # Populate the combo box with the enumeration elements, use the form context for translation
+            elements = [translate(self.form.objectName(), custom_lbls.get(itm, itm)) for itm in enum]
+            widget.clear()
+            widget.addItems(elements)
+
+            def _getter():
+                return enum[widget.currentIndex()]
+
+            def _setter(val):
+                widget.setCurrentIndex(enum.index(val))
+
+            self._fc[prop_name] = _getter, _setter
+        elif prop_type == "App::PropertyBool" and widget_type == "QCheckBox":
+            self._fc[prop_name] = widget.isChecked, widget.setChecked
+        elif prop_type in PROP_TYPE_NUMERIC and widget_type == "QDoubleSpinBox":
+            self._fc[prop_name] = widget.value, widget.setValue
+        elif prop_type in PROP_TYPE_QTYES and widget_type == "QLineEdit":
+            self._fc[prop_name] = widget.text, lambda v: widget.setText(str(v))
+        else:
+            raise ValueError(f"Unsupported connection between '{prop_type}' property and '{widget_type}' widget")
+        # Set the tooltip to the one corresponding to the property.
+        widget.setToolTip(translate("App::Property", self.obj.getDocumentationOfProperty(prop_name)))
+
```

Comment:
I will add docs and move this to a new file, Id like to know what you guys think of this approach.